scripts/check_compatibility_careamics.py

```diff
@@ -0,0 +1,198 @@
+from typing import Protocol, Optional, Union, Tuple, List
+import argparse
+from pathlib import Path
+import traceback
+from functools import lru_cache
+
+import pydantic
+from careamics import __version__ as CAREAMICS_VERSION
+from careamics import CAREamist
+from careamics.lightning import FCNModule, VAEModule
+from careamics.config import Configuration
+from careamics.model_io.bmz_io import load_from_bmz
+from bioimageio.spec import load_model_description
+from bioimageio.spec._internal.url import HttpUrl
+from bioimageio.core.digest_spec import get_test_inputs
+from bioimageio.spec.model import AnyModelDescr
+from bioimageio.spec.model.v0_5 import ModelDescr, AxisId
+
+from .script_utils import CompatibilityReportDict, check_tool_compatibility
+
+
+@lru_cache
+def careamics_load_from_bmz(
+    rdf_url: Union[HttpUrl, pydantic.DirectoryPath],
+) -> Tuple[Union[FCNModule, VAEModule], Configuration]:
+    return load_from_bmz(rdf_url)
+
+
+class CompatibilityCheck_v0_5(Protocol):
+
+    def __call__(
+        self, model_desc: ModelDescr, rdf_url: str
+    ) -> Optional[CompatibilityReportDict]: ...
+
+
+def check_model_desc_v0_5(
+    model_desc: AnyModelDescr,
+) -> Optional[CompatibilityReportDict]:
+    if not isinstance(model_desc, ModelDescr):
+        return CompatibilityReportDict(
+            status="failed",
```

Comment:
Yeah CAREamics doesn't create any v0.4 models so this check acts as a filter which results in the type narrowing you suggested earlier, to simplify the checks.

Reply:
hmm actually, I suppose someone could tag a non careamics v0.4 model as careamics and then there wouldn't be any warning that it's not a careamics supported model...

Reply:
it will not get the `ceramics-compatible` tag though. 
we can then list all models tagged with `ceramics` and not `ceramics-compatible`.
Do you agree that for the first version `not-applicable` is fine? You can refine the checks anytime.

Reply:
Yes, I think that should be fine, thanks!